### DISCUSSION TITLE
How to use Mapbox as tile provider?

### DISCUSSION THREAD

**Discussion by comatory** [fleaflet/flutter_map#1028] [Q&A] [answered]
Hello. I already raised the question [here](https://github.com/fleaflet/flutter_map/issues/949) but got no answers.

The documentation does not specify how to use Mapbox really and docs on Mapbox page are not relevant. I ended up using something like this

`https://api.mapbox.com/styles/v1/myAccount/abcd1234/tiles/256/{z}/{x}/{y}@2x`

But there are few problems with this:

1. I have to create tileset in Mapbox studio even though I'm not planning on using any custom tiles. I just want to use their hosted tiles.
2. For some strange reason this approach which I'm not sure is because of Mapbox I can't zoom all the way in. Once I  do it, the tiles disappear at certain zoom level.

I would appreciate any tips. Perhaps I'm doing something completely weird but I consider Mapbox to be quite popular choice and it's a shame there's no mention of it in documentation.

P.S. No I don't want to use Google Maps Flutter plugin for specific reason. I don't want to use `mapbox_gl` as it's quite low level and I simply find `flutter_map` to be easiest to work with.

**Comment by JaffaKetchup** [comment]
>For some strange reason this approach which I'm not sure is because of Mapbox I can't zoom all the way in. Once I do it, the tiles disappear at certain zoom level.

Can you say what zoom level the tiles disappear at? Zoom level 18 or 19 as maximum is quite normal.

**Comment by comatory** [comment]
On Mon Sep 13, 2021 at 8:01 AM CEST, Luka S wrote:
> >For some strange reason this approach which I'm not sure is because of Mapbox I can't zoom all the way in. Once I do it, the tiles disappear at certain zoom level.
>
> Can you say what zoom level the tiles disappear at? Zoom level 18 or 19
> as maximum is quite normal.

It's at zoom level 18.0. It's zoomed in pretty nicely so it's not a big
deal I just wondered why the tiles disappear.

I don't see into specifics and I'm not familiar with how mapping
libraries work but I assumed the received tiles are vectors so
theoretically, there should be no limit.

**Reply by JaffaKetchup** [reply]
No, I don't think the tiles are vector tiles because if they were, you'd need to use an external plugin for `flutter_map`, like `vector_map_tiles` by @greensopinion.
You still might be able to get vector tiles and use them with this plugin, but if you're not using that plugin yet, then those tiles are raster.
Therefore it makes sense that the maximum limit is 18. If you don't want to see the grey tiles, set the `maxZoom:` (I think it's in `MapOptions`).
Hope that helped!

**Comment by comatory** [comment]
On Mon Sep 13, 2021 at 9:01 AM CEST, Luka S wrote:
> Therefore it makes sense that the maximum limit is 18. If you don't want
> to see the grey tiles, set the `maxZoom:` (I think it's in
> `MapOptions`).
> Hope that helped!

Yep that's what I'm doing. Did not know about the vector tile plugin,
thanks.

However the question still stands. How to use public Mapbox tiles?

**Comment by ibrierley** [comment]
You can use public Mapbox tiles no problem with a key, (there issue is more
often than not related to what style). Maybe you could expand on what style
you are wanting.

**Comment by comatory** [comment]
For example, I'm trying to use `light-v10` tiles provided via this URL: `https://api.mapbox.com/styles/mapbox/light-v10`

I tried this template URL: `https://api.mapbox.com/styles/mapbox/light-v10/tiles/256/{z}/{x}/{y}@2x` but tiles would not load for me.

**Comment by ibrierley** [comment]
I'm not sure light-v10 is a public available tileset (for raster, I think thats vector, it's possible it's both though...) (I may be wrong though, Mapbox are VERY vague on what's available and how to use sometimes)....
Do you have an example of it being used somewhere as a raster source ?

**Comment by comatory** [comment]
On Mon Sep 13, 2021 at 10:38 AM CEST, Ian wrote:
> I'm not sure light-v10 is a public available tileset (for raster, I
> think thats vector, it's possible it's both though...) (I may be wrong
> though, Mapbox are VERY vague on what's available and how to use
> sometimes)....
> Do you have an example of it being used somewhere as a raster source ?

Honestly up until now I did not differentiate between vector/raster
like I mentioned above. I have used this specific tileset with
their Javascript SDK before so not Flutter.

Do they have a list of public (raster) tilesets? Also I don't mind
using vector tiles, what would the vector template URL look
like?

EDIT: I did find [this](https://docs.mapbox.com/api/maps/styles/#mapbox-styles) link
but it does not say which is vector or raster.

I think in Mapbox studio light style is available but I do not want
to use my own styles I just want to use the available Mapbox styles.

**Comment by ibrierley** [comment]
How about this...

https://api.mapbox.com/styles/v1/mapbox/light-v10/tiles/1/1/0?access_token={your
token id}

**Comment by comatory** [comment]
@ibrierley it does render something:

![01](https://user-images.githubusercontent.com/5990424/133055565-c752000a-9e20-4dac-be6c-36367894d846.jpg)

Zooming in doesn't really work, it seems messed up (the blue dot showing current location is not correct). Wouldn't the template need `{x} {y} {z}` parameters?

**Comment by comatory** [comment, marked as answer]
Oh yeah that's right template should be `https://api.mapbox.com/styles/v1/mapbox/light-v10/tiles/{z}/{x}/{y}`.

That seems to be correct template URL. Thanks for helping me with this.

**Comment by ibrierley** [comment]
No problem, Mapbox could be a bit more user friendly with some better examples for devs, and a better description of what's available tbh.

**Comment by comatory** [comment]
I opened a PR to improve the docs: https://github.com/fleaflet/flutter_map/pull/1030

**Comment by sharifhimu** [comment]
For me, I was trying to load mapbox url in flutter, 
https://api.mapbox.com/styles/v1/mapbox/light-v10/tiles/1/1/0?access_token={your
token id}
was working for me. But, I wanted to load 3d-terrain. which I am not able to load. 
Can anyone tell me what should be the url for 3d-terrain rendering?
